modules/extensions-support/src/test/java/org/mule/runtime/module/extension/internal/loader/validation/NameClashModelValidatorTestCase.java

```diff
@@ -921,6 +924,27 @@ public void contentParametersWithSameNameNoTypeIdButSameType() {
     validate();
   }
 
+  @Test
+  public void classLoaderIsNoLongerUsedToValidateSingularizedChildNames() throws ClassNotFoundException {
+    ParameterModel plural = getParameter(CHILD_PLURAL_PARAM_NAME, childTestList);
+    ParameterModel singular = getParameter(CHILD_SINGULAR_PARAM_NAME, ChildTest.class);
+    when(constructModel.getAllParameterModels()).thenReturn(asList(singular, plural));
+    validate();
+    verify(extensionModel, times(0)).getModelProperty(ClassLoaderModelProperty.class);
+  }
+
+  @Test
+  public void classLoaderIsNoLongerUsedToValidateContentNamesMatchType() {
+    ParameterModel firstParam = getParameter(CHILD_SINGULAR_PARAM_NAME, Pojo.class);
+    when(firstParam.getRole()).thenReturn(PRIMARY_CONTENT);
+    ParameterModel secondParam = getParameter(CHILD_SINGULAR_PARAM_NAME, Pojo.class);
+    when(secondParam.getRole()).thenReturn(CONTENT);
+    when(operationModel.getAllParameterModels()).thenReturn(asList(firstParam));
+    when(sourceModel.getAllParameterModels()).thenReturn(asList(secondParam));
+    validate();
+    verify(extensionModel, times(0)).getModelProperty(ClassLoaderModelProperty.class);
```

Comment:
Instead of verifying that the model property is retrieved or not, use metadata types with class information annotation that point to classes that do not exist, and assert that there is are no errors in the problemReporter